File: packages/es-markdown-core/src/model/cross-embedded.ts
import type { MarkupBase } from "./base.js";
import type { Destination, Dimensions } from "../values.js";

/** A workspace transclusion. The label is absent only when no separator was written. */
export interface CrossEmbedded extends MarkupBase<"crossEmbedded"> {
    readonly dest: Destination;
    /** Raw prefix after a valid dimension suffix; null only when no separator was authored. */
    readonly label: string | null;
    /** Authored dimensions, absent for an invalid or missing suffix. */
    readonly dimensions: Dimensions | null;
}

File: packages/kotlin-markdown-core/src/commonMain/kotlin/com/nouprax/markdown/core/model/CrossEmbedded.kt
package com.nouprax.markdown.core

/** A workspace transclusion. [label] is absent only when no separator was written. */
public class CrossEmbedded internal constructor(
    public val dest: Destination,
    /** Raw prefix after a valid size suffix. */
    public val label: String?,
    /** Authored dimensions, absent for an invalid or missing suffix. */
    public val dimensions: Dimensions?,
    override val scope: Scope,
    override val anchor: String?,
    override val attributes: Attributes,
) : Markup {
    init {
        require(dimensions == null || label != null) { "dimensions require an authored label" }
    }

    override fun <Result> accept(visitor: Visitor<Result>): Result = visitor.visit(this)
}
